samigo/samigo-app/src/java/org/sakaiproject/tool/assessment/ui/listener/delivery/BeginDeliveryActionListener.java

```diff
@@ -174,6 +181,19 @@ public void processAction(ActionEvent ae) throws
     delivery.setFileUploadSizeMax(Math.round(sizeMax.floatValue()/1024));
     delivery.setPublishedAssessment(pub);
 
+    if (secureDelivery.isSecureDeliveryAvaliable()) {
+      String secureDeliveryModuleId = pub.getAssessmentMetaDataByLabel(SecureDeliveryServiceAPI.MODULE_KEY);
+
+      if (StringUtils.equals(secureDeliveryModuleId, SecureDeliverySeb.MODULE_NAME)) {
+        delivery.setSebSetup(true);
+        delivery.setSecureDeliveryHTMLFragment(secureDelivery.getHTMLFragment(secureDeliveryModuleId, pub,
+            null, Phase.ASSESSMENT_START, PhaseStatus.FAILURE, null));
+        PersistenceService persistenceService = PersistenceService.getInstance();
+        persistenceService.getSebValidationFacadeQueries()
+            .expireSebValidations(pub.getPublishedAssessmentId(), AgentFacade.getAgentString());
```

Comment:
why is SEB-specific code here?

the module-specific code should be in setSecureDeliveryHTMLFragment